app/controllers/discourse_login_client/auth_controller.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module DiscourseLoginClient
+  class AuthController < ApplicationController
+    requires_plugin "discourse-login-client"
+
+    skip_before_action :verify_authenticity_token, only: [:revoke]
+
+    def revoke
+      signature = params.require(:signature)
+      user_id = params.require(:user_id)
+      timestamp = params.require(:timestamp)
+
+      RateLimiter.new(nil, "discourse_login_revoke_#{user_id}", 5, 1.minute).performed!
+
+      time_diff = (Time.now.to_i - timestamp.to_i).abs
+      if time_diff > 5.minutes.to_i
+        if SiteSetting.discourse_login_debug_auth
+          Rails.logger.warn("Expired timestamp in discourse_login revoke: #{time_diff} seconds old")
+        end
+        return render_invalid_request
+      end
+
+      return render_invalid_request if (client_id = SiteSetting.discourse_login_client_id).blank?
+      return render_invalid_request if (client_secret = SiteSetting.discourse_login_client_secret).blank?
+
+      hashed_secret = Digest::SHA256.hexdigest(client_secret)
+
+      expected_signature =
+        OpenSSL::HMAC.hexdigest("sha256", hashed_secret, "#{client_id}:#{user_id}:#{timestamp}")
+
+      if !ActiveSupport::SecurityUtils.secure_compare(signature, expected_signature)
+        if SiteSetting.discourse_login_debug_auth
+          Rails.logger.warn("Invalid signature for user_id #{user_id} in discourse_login revoke")
+        end
+        return render_invalid_request
+      end
+
+      unless uaa = UserAssociatedAccount.find_by(provider_name: "discourse_login", provider_uid: user_id)
+        if SiteSetting.discourse_login_debug_auth
+          Rails.logger.warn("User not found with provider_uid: #{user_id}")
+        end
+        return render_invalid_request
+      end
+
+      UserAuthToken.where(user_id: uaa.user_id).destroy_all
```

Comment:
The suggestion in the other PR to use `user_secure_identifier` would avoid any confusion with `uaa.user_id` here.